pkgs/applications/misc/onboard/default.nix

```diff
@@ -0,0 +1,133 @@
+  src = fetchurl {
+    url = "https://launchpad.net/onboard/${majorVersion}/${version}/+download/${name}.tar.gz";
+    sha256 = "01cae1ac5b1ef1ab985bd2d2d79ded6fc99ee04b1535cc1bb191e43a231a3865";
+  };
+
+  doCheck = false;
+
+  propagatedBuildInputs = [
+    gtk3
+    libcanberra_gtk3
+    libudev
+    bash
+    hunspell
+    isocodes
+    pkgconfig
+    xorg.libXtst
+    xorg.libxkbfile
+    libxkbcommon
+    intltool
+    python3
+    python3.pkgs.pycairo
+    python3.pkgs.dbus-python
+    python3.pkgs.pygobject3
+    python3.pkgs.systemd
+    python3.pkgs.distutils_extra
+    python3.pkgs.pyatspi
+    glib
+  ];
+
+  buildInputs = [
+    glib gobjectIntrospection gsettings_desktop_schemas gnome3.dconf wrapGAppsHook
+  ] ++ stdenv.lib.optional atspiSupport at_spi2_core;
+
+  preBuild = ''
+    rm -r Onboard/pypredict/attic
+
+    substituteInPlace  ./scripts/sokSettings.py \
+    --replace "#!/usr/bin/python3" "" \
+    --replace "PYTHON_EXECUTABLE," "\"$out/bin/onboard-settings\"" \
+    --replace '"-cfrom Onboard.settings import Settings\ns = Settings(False)"' ""
+
+    chmod -x ./scripts/sokSettings.py
+
+    for dir in ". ./Onboard/ ./scripts/ ./Onboard/test/ ./Onboard/pypredict/
+    ./Onboard/pypredict/lm/ ./Onboard/pypredict/test/ ./Onboard/pypredict/tools/";
+    do patchShebangs $dir; done
+
+    substituteInPlace  ./Onboard/LanguageSupport.py \
+    --replace "/usr/share/xml/iso-codes" "${isocodes}/share/xml/iso-codes" \
+    --replace "/usr/bin/yelp" "${yelp}/bin/yelp"
+
+    substituteInPlace  ./Onboard/Indicator.py \
+    --replace   "/usr/bin/yelp" "${yelp}/bin/yelp"
+
+    substituteInPlace  ./gnome/Onboard_Indicator@onboard.org/extension.js \
+    --replace "/usr/bin/yelp" "${yelp}/bin/yelp"
+
+    substituteInPlace  ./Onboard/SpellChecker.py \
+    --replace "/usr/share/hunspell" ${hunspell}/bin/hunspell \
+    --replace "/usr/lib" "$out/lib"
+
+    substituteInPlace  ./data/org.onboard.Onboard.service  \
+    --replace "/usr/bin" "$out/bin"
+
+    substituteInPlace  ./Onboard/utils.py \
+    --replace "/usr/share" "$out/share"
+    substituteInPlace  ./onboard-defaults.conf.example \
+    --replace "/usr/share" "$out/share"
+    substituteInPlace  ./Onboard/Config.py \
+    --replace "/usr/share/onboard" "$out/share/onboard"
+
+    substituteInPlace  ./Onboard/WordSuggestions.py \
+    --replace "/usr/bin" "$out/bin"
+
+    substituteInPlace  ./setup.py \
+    --replace "/bin/bash" ${stdenv.shell}
+    substituteInPlace  ./Onboard/TextDomain.py \
+    --replace "/bin/bash" "${stdenv.shell}"
+  '';
+
+  postInstall = ''
+    mkdir -p $out/share/glib-2.0/schemas/ $out/lib/girepository-1.0
+    cp onboard-default-settings.gschema.override.example $out/share/glib-2.0/schemas/10_onboard-default-settings.gschema.override
+
+    ${glib.dev}/bin/glib-compile-schemas $out/share/glib-2.0/schemas/
+
+    addToSearchPath GI_TYPELIB_PATH $out/lib/girepository-1.0
+    addToSearchPath XDG_DATA_DIRS $out/share
```

Comment:
Similarly here, `addToSearch` adds to path of packages that use this dependency. And I think it would only work as a part of setup hook anyway.